cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -437,6 +454,44 @@ private void performUpgrade() throws Exception {
     }
   }
 
+  private void upgradeMetadataDatasetSpecs() {
+    upgradeMetadataDatasetSpec(DefaultMetadataStore.BUSINESS_METADATA_INSTANCE_ID);
+    upgradeMetadataDatasetSpec(DefaultMetadataStore.SYSTEM_METADATA_INSTANCE_ID);
+  }
+
+  private void upgradeMetadataDatasetSpec(DatasetId metadataDatasetId) {
+    DatasetSpecification oldBusinessMetadataSpec = datasetInstanceManager.get(metadataDatasetId);
```

Comment:
nit: variable name should have business since the method is called for system too.